teacher/view_attendance.html

```diff
@@ -0,0 +1,177 @@
+        sname = document.createElement("td");
+        sgrade = document.createElement("td");
+        shomeroom = document.createElement("td");
+        sname.style.width = "200px";
+        sgrade.style.width = "16px";
+        shomeroom.style.width = "20px";
+        sname.innerHTML = "{{s['first']}} {{s['last']}}";
+        sgrade.innerHTML = "{{s['current_grade']}}";
+        shomeroom.innerHTML = "{{s['current_homeroom']}}";
+        row.append(sname, sgrade, shomeroom);
+ 
+        {% for d in c['dates_sorted'] %}
+          td = document.createElement("td");
+          {% if s['email'] in c['attendance'][d[0]]['absent'] %}
+            td.innerHTML = "X";
+            td.style.background = ABSENT_COLOR;
+            td.title = "{{c['attendance'][d[0]]['note']}}";
+          {% elif s['email'] in c['attendance'][d[0]]['present'] %}
+            td.innerHTML = "\u2713";
+            td.style.background = PRESENT_COLOR;
+            td.title = "{{c['attendance'][d[0]]['note']}}";
+          {% else %}
+            td.innerHTML = " ";
+            td.title = "{{c['attendance'][d[0]]['note']}}";
+          {% endif %}
+          td.style.textAlign = "center";
+          row.append(td);
+        {% endfor %}
+        tbody.appendChild(row);
+      {% endfor %}
+
+      table.appendChild(tbody);
+      class_div.appendChild(table);
+      absences.appendChild(class_div);
+    {% endif %}
+  {% endfor %}
+}
+
+function fillDaypartDropDown() {
+  let daypart_dd = document.getElementById("daypartDropDown");
+  let opt = document.createElement("option");
+  opt.text = "All";
+  opt.selected = "selected";
+  daypart_dd.add(opt);
+
+  {% for dp in dayparts %}
+    opt = document.createElement("option");
+    opt.text = "{{dp['name']}}";
+    {% if dp['name'] == selected_daypart %}
+      opt.selected = "selected";
+    {% endif %}
+    daypart_dd.add(opt);
+  {% endfor %}
+}
+
+function sendGet(params) {
+  let form = document.createElement("form");
+  form.setAttribute("method", "get");
+  for (let key in params) {
+    if (params.hasOwnProperty(key)) {
+      let hiddenField = document.createElement("input");
+      hiddenField.setAttribute("type", "hidden");
+      hiddenField.setAttribute("name", key);
+      hiddenField.setAttribute("value", params[key]);
+      form.appendChild(hiddenField);
+    }
+  }
+  document.body.appendChild(form);
+  form.submit();
+}
+
+window.onload = function() {
+  let start_input = document.getElementById("startDate");
+  let end_input = document.getElementById("endDate");
+  let daypart_dd = document.getElementById("daypartDropDown");
+
+  fillDaypartDropDown();
+  displayAttendance();
+
+  daypart_dd.onchange = function() {
+    sendGet({"institution" : "{{institution}}",
+             "session" : "{{session}}",
+             "selected_daypart": daypart_dd.value});    
+  }
+}
+</script>
+
+{% if not teacher %}
+Teacher not found! This error should never happen. Please inform the selectives team. Thanks!
```

Comment:
how to inform the team? maybe let's add an email link?

Reply:
Good idea. I'll add an email link.